userdocs/diagnostics/actor-isolated-call.md

```diff
@@ -0,0 +1,46 @@
+# Calling an actor-isolated method from a synchronous nonisolated context
+
+Calls to actor-isolated methods from outside the actor must be done asynchronously. Otherwise, access to actor state can happen concurrently and lead to data races. These rules also apply to global actors like the main actor.
+
+For example:
+
+```swift
+@MainActor
+class MyModel {
+  func update() { ... }
+}
+
+func runUpdate(model: MyModel) {
+  mode.update()
+}
+```
+
+Building the above code produces an error about calling a `@MainActor`-isolated method from outside the actor:
+
+```
+| func runUpdate(model: MyModel) {
+|   model.update()
+|         `- error: call to main actor-isolated instance method 'update()' in a synchronous nonisolated context
+| }
+```
+
+The `runUpdate` function doesn't specify any actor isolation, so the default is `nonisolated`, meaning it can be called from anywhere.
```

Comment:
```suggestion
The `runUpdate` function doesn't specify any actor isolation, so the default is `nonisolated`, which implies to the compiler that it can be called from anywhere.
```

Reply:
I feel like language like "implies to the compiler" makes it sound too much like we just need to appease the compiler. I'd rather elaborate more on what `nonisolated` means semantically and how data races are prevented - `runUpdate` can be called from any concurrency domain, and for that to be safe from data races, the implementation must not access any actor isolated state.